streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -342,7 +342,7 @@ public void flush(final InternalProcessorContext context) {
     /**
      * @throws IOException if any error happens when closing the state stores
      */
-    public void close(Map<TopicPartition, Long> ackedOffsets) throws IOException {
+    public void close(Map<TopicPartition, Long> ackedOffsets, final boolean writeCheckpoint) throws IOException {
```

Comment:
@dguy As discussed above, could we let `AbstractTask` to simply pass null if its `writeCheckpoint` flag is false, and otherwise pass `recordCollectorOffsets()`? Then in this class we can simply check on the passed `ackedOffsets` not null to write the file.